icepyx/core/query.py

```diff
@@ -392,6 +394,10 @@ class Query(GenQuery, EarthdataAuthMixin):
     GenQuery
     """
 
+    _CMRparams: apifmt.CMRParameters
+    _reqparams: apifmt.RequiredParameters
+    _subsetparams: apifmt.SubsetParameters | None
```

Comment:
What is this doing?

Reply:
This is specifying the type of these class attributes. So if we were do to `self._reqparams = 5`, the typechecker would yell at us for violating this type specification.

Reply:
> This is specifying the type of these class attributes. 

I thought this was happening below in (e.g. line 540) `def CMRparams(self) -> CMRParams:`, but missed the distinction between typing the public and private versions of `CMRparams`. Thanks for clarifying!